src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateAtTheMoment = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    let stateCopy = { ...stateAtTheMoment };
```

Comment:
Instead of creating 2 copies of state, let's keep one and when you are pushing it to the history use spread operator `stateHistory.push({...stateCopy})`